### PR TITLE
Copy LICENSE files when building packages

### PR DESCRIPTION
**Pull request by joshwooding**
Part of #2160

### CODE REVIEW

**Comment by cristiano-belloni** [comment]
@joshwooding sorry for the delay, will take a look at this shortly in the next days

File: packages/modular-scripts/src/build/buildPackage/maybeCopyRootLicense.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+export async function maybeCopyRootLicense(
+  target: string,
+  targetOutputDirectory: string,
+) {
+  const logger = getPrefixedLogger(target);
+  const modularRoot = getModularRoot();
+
+  const targetOutputLicense = path.join(targetOutputDirectory, 'LICENSE');
+  if (!fs.existsSync(targetOutputLicense)) {
```
Comment (benpryke) [inline packages/modular-scripts/src/build/buildPackage/maybeCopyRootLicense.ts:14]: Sometimes people use `LICENSE.txt`, `LICENSE.md` etc. Maybe we should copy any file of the form `LICENSE.*`?

**Comment by cristiano-belloni** [comment]
LGTM, I think this is ready to merge after addressing @benpryke's feedback.

File: packages/modular-scripts/src/build/buildPackage/maybeCopyRootLicense.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  target: string,
+  targetOutputDirectory: string,
+) {
+  const logger = getPrefixedLogger(target);
+  const modularRoot = getModularRoot();
+  const matches = globby.sync(path.join(targetOutputDirectory, licenseGlob), {
+    cwd: modularRoot,
+    onlyFiles: true,
+  });
+  if (matches.length == 0) {
```
Comment (benpryke) [inline packages/modular-scripts/src/build/buildPackage/maybeCopyRootLicense.ts:19]: nit: `===`

Comment (benpryke) [reply packages/modular-scripts/src/build/buildPackage/maybeCopyRootLicense.ts:19]: If the target package contains a license file, would this have been copied over by this point?

File: packages/modular-scripts/src/build/buildPackage/maybeCopyRootLicense.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import globby from 'globby';
+import getPrefixedLogger from '../../utils/getPrefixedLogger';
+import getModularRoot from '../../utils/getModularRoot';
+
+const licenseGlob = 'LICEN@(C|S)E*';
+
+export async function maybeCopyRootLicense(
+  target: string,
```
Comment (benpryke) [inline packages/modular-scripts/src/build/buildPackage/maybeCopyRootLicense.ts:10]: Should we not check for a license file in the target directory first?

File: packages/modular-scripts/src/build/buildPackage/maybeCopyRootLicense.ts (hunk 2 above)
Comment (joshwooding) [reply packages/modular-scripts/src/build/buildPackage/maybeCopyRootLicense.ts:19]: Yeah, due to pack-list and the default files array containing LICENSE. It's worth adding that if a user is using a different license format they would have to add it to files manually.

Comment (joshwooding) [reply packages/modular-scripts/src/build/buildPackage/maybeCopyRootLicense.ts:19]: Unless you think modular should try and detect the license in that case too?

**Comment by joshwooding** [comment]
Are there any more changes needed here?

**Comment by joshwooding** [comment]
@cristiano-belloni Do you know if this PR can be merged too

**Pull request merged**